api/v1beta1/azuremachine_webhook.go

```diff
@@ -136,6 +136,13 @@ func (m *AzureMachine) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
+	if !reflect.DeepEqual(m.Spec.Diagnostics, old.Spec.Diagnostics) {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "diagnostics"),
+				m.Spec.Diagnostics, "field is immutable"),
```

Comment:
Why is this? Because we won't update it on the actual VMs?

Reply:
Yes, and the other fields at this level have this too.